src/model/balance.js

```diff
@@ -12,59 +12,43 @@ module.exports = class Balance extends EventEmitter {
     super()
 
     this._maximum = new BigNumber(opts.maximum)
-    this._store = opts.store
-    this._cached = null
-
-    // all reserved DB key prefixes are 9 characters
-    // so it lines up with 'transfer_'
-    this._key = 'balance__'
+    this._balance = null
   }
 
-  * get () {
-    return (yield this._getNumber()).toString()
+  set (balance) {
+    this._balance = new BigNumber(balance)
```

Comment:
you assert in the other functions that the passed in parameter is a number. Any reason you don't do it here?